### PR TITLE
Feat: Added Shoelace algorithm

### PR DESCRIPTION
**Pull request by Renjian-buchai**
#### Description of Change

Feat: Added shoelace algorithm 

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: math/iterative_factorial.cpp [hunk 1]
```diff
@@ -0,0 +1,116 @@
+/**
+ * @file
+ * @brief Iterative implementation of [Factorial](https://en.wikipedia.org/wiki/Factorial)
+ *
+ * @author [Renjian-buchai](https://github.com/Renjian-buchai)
```
Comment (realstealthninja) [inline math/iterative_factorial.cpp:5]: bit of a nit pick but would you mind moving the author tag to the bottom as is tradition

File: math/iterative_factorial.cpp [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+    // Exception test
+    std::cout << "Exception test \n"
+                 "Input: 21 \n"
+                 "Expected output: Exception thrown \n";
+    try {
+        math::iterativeFactorial(21);
+    } catch (std::invalid_argument& e) {
+        std::cout << "Exception thrown successfully \n";
+    }
```
Comment (realstealthninja) [inline math/iterative_factorial.cpp:106]: please add a print statement saying all tests have passed;
```suggestion
    }
    std::cout << ""all tests have passed\n"
```

File: math/shoelace_algorithm.cpp [hunk 3]
```diff
@@ -0,0 +1,141 @@
+/**
+ * @file
+ * @brief Implementation of the [shoelace
+ * algorithm](https://en.wikipedia.org/wiki/Shoelace_formula)
+ *
+ * @author [Renjian-buchai](https://github.com/Renjian-buchai)
+ *
+ * @details Calculates area of a polygon using the coordinates of its vertices.
+ * Area of polygon
+ * = 0.5 | (\x1, \y1) (\x2, \y2) ... (\xn, \yn) (\x1, \y1) |
```
Comment (realstealthninja) [inline math/shoelace_algorithm.cpp:10]: i think you need to add \f$ for \x1 to be recognized.

File: math/shoelace_algorithm.cpp [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+ * matter whether the points are inputted in clockwise or counterclockwise
+ * order.
+ */
+
+#include <array>     /// For std::array
+#include <cassert>   /// For assert
+#include <cmath>     /// For std::abs
+#include <cstdint>   /// For int64_t
+#include <iostream>  /// For std::cout
+
```
Comment (realstealthninja) [inline math/shoelace_algorithm.cpp:67]:
```suggestion
/**
 * @namespace
 * @brief Mathematical algorithms
 */
```

File: math/shoelace_algorithm.cpp [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+}
+
+/**
+ * @brief Main function
+ * @return 0 on exit
+ */
+int main() {
+    test();  // Run self-test implementation
+    return 0;
+}
```
Comment (realstealthninja) [inline math/shoelace_algorithm.cpp:141]:
```suggestion
}

```

File: math/shoelace_algorithm.cpp [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+                                      array<int64_t, 2>{8, 6}};
+    assert(math::shoelace(case3) == 7);
+
+    std::cout << "Case 4 (Clockwise):\n"
+                 "Input: | (-2, 4) (-2, 1) (3, -3) (4, 4) |\n"
+                 "Expected output: 28.5\n";
+    array<array<int64_t, 2>, 4> case4{
+        array<int64_t, 2>{-2, 4}, array<int64_t, 2>{-2, 1},
+        array<int64_t, 2>{3, -3}, array<int64_t, 2>{4, 4}};
+    assert(math::shoelace(case4) == 28.5);
```
Comment (realstealthninja) [inline math/shoelace_algorithm.cpp:131]:
```suggestion
    assert(math::shoelace(case4) == 28.5);
    std::cout << "all tests have passed"
```

File: math/shoelace_algorithm.cpp [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+ */
+template <typename T, size_t N>
+double shoelace(std::array<std::array<T, 2>, N> coordinates) {
+    double sum = 0;
+    // Cross-multiplies and sums up everything from \x1 to \x(n-1)
+    for (size_t i = 0; i < N - 1; ++i) {
+        sum += coordinates[i][0] * coordinates[i + 1][1] -
+               coordinates[i + 1][0] * coordinates[i][1];
+    }
+    // Cross multiplies \xn and \x1
```
Comment (realstealthninja) [inline math/shoelace_algorithm.cpp:86]: i dont think documentation tags work in comments

**Review by realstealthninja — CHANGES_REQUESTED** [review]

File: math/iterative_factorial.cpp [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+ *
+ * @details Calculates factorial iteratively.
+ * \f[n! = n\times(n-1)\times(n-2)\times(n-3)\times\ldots\times3\times2\times1
+ *       = n\times(n-1)!\f]
+ * for example:
+ * \f$4! = 4\times3! = 4\times3\times2\times1 = 24\f$
+ *
+ * @example
+ *
+ * 5! = 5 * 4 * 3 * 2 * 1
```
Comment (realstealthninja) [inline math/iterative_factorial.cpp:15]: Could you turn this into a LaTex formating

**Review by realstealthninja — CHANGES_REQUESTED** [review]

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Please create separate PRs for each file.
This will make it easier to review one as best as possible. 🙂

**Pull request closed without merging**